### PR TITLE
Initial version of R-Car weston images

### PR DESCRIPTION
**Pull request by andr2000**

### CODE REVIEW

File: classes/xt_quircks.bbclass [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    if type.lower() != 'file':
+        return url
+    if not os.path.isdir(location):
+        return url
+    new_location = create_tarfile(location, d)
+    return url.replace(location, new_location)
+
+def remove_tarfile(url, type, location, d):
+    if type.lower() != 'file':
+        return url
```
Comment (lorc) [inline classes/xt_quircks.bbclass]: probably just `return` ?

File: classes/xt_quircks.bbclass [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    if not os.path.isdir(location):
+        return url
```
Comment (lorc) [inline classes/xt_quircks.bbclass]: the same. Should you return `url` there?

File: classes/xt_quircks.bbclass [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    arcname = get_tarfname(location, d)
+    if os.path.exists(arcname):
+        os.remove(arcname)
+
+addtask unpack_xt_extras after do_unpack before do_configure
+python do_unpack_xt_extras() {
+    bb.debug(1, "Unpacking Xen-troops extras")
+    # all the sources in SRC_URI have been unpacked already, clean it
+    d.setVar("SRC_URI", "")
```
Comment (lorc) [inline classes/xt_quircks.bbclass]: Can't this broke some other target? E.g. there will some other `bbclass` that relies on `SRC_URI` at this stage.

**Review by lorc — CHANGES_REQUESTED** [review]

File: classes/xt_quircks.bbclass (hunk 1 above)
Comment (andr2000) [reply classes/xt_quircks.bbclass]: ok

File: classes/xt_quircks.bbclass (hunk 2 above)
Comment (andr2000) [reply classes/xt_quircks.bbclass]: ok

File: classes/xt_quircks.bbclass (hunk 3 above)
Comment (andr2000) [reply classes/xt_quircks.bbclass]: as per my understanding Yocto/bitbake do not use global environment, e.g. the "d" variable, which represents current execution thread's own environment, has no visibility to other's threads env and vise versa. Thus, I consider this safe.
What is more, this "d" variable changes do not even survive between different tasks of the same recipe.
Won't fix

**Comment by lorc** [comment]
Thank you. Looks fine. 

`Reivewed-by Volodymyr Babchuk <vlad.babchuk@gmail.com>

**Review by aanisov — APPROVED** [review]

**Pull request merged**